### ISSUE TITLE
Move login/logout logic into the useUser composable and update references

### ISSUE THREAD

**Issue by rtibbles** [learningequality/kolibri#12204]
### Prerequisites
- [x] https://github.com/learningequality/kolibri/issues/12203

The [login, logout, and session actions](https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/state/modules/core/actions.js#L95) should be migrated into the useUser composable.

As this will have been done after the migration of all references to the getters and state to the useUser composable, the state should be removed from Vuex state and made into module level state in the useUser composable. The getters should be transferred to the useUser composable as computed refs based on the module level state.

Any references to the `CORE_SET_SESSION` mutation should also be updated: https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/state/modules/session.js#L89

**Comment by iamshobhraj** [comment]
Hi @rtibbles , can i get this assigned?

**Comment by rtibbles** [comment]
Yes, certainly! Please feel free to ask any questions about this, and also look at previous related PRs for guidance!

**Comment by iamshobhraj** [comment]
hi @rtibbles , i opened a pr for this issue. i moved the login/logout logic along with session logic in `useUser` composable but when i tried to update the `CORE_SET_SESSION`, some tests failed. I  am unable to figure out why.

**Comment by MisRob** [comment]
Hi @iamshobhraj, I wanted to mention that [Learning Equality will be closed from December 23 to January 5](https://github.com/learningequality/kolibri/discussions/12956).

**Comment by yashhash2** [comment]
Hi @iamshobhraj are you still working on this issue if no then @rtibbles can assign it to me, if you are still working then all the best :)

**Comment by rtibbles** [comment]
There is some very strong work in progress here, that I am hopeful @iamshobhraj will come back to. If not, we will finish up this task.

**Comment by iamshobhraj** [comment]
Hi @rtibbles, I’ve decided to close the PR #12915  for now as I need more time to investigate and resolve the issue I was facing. In the meantime, please feel free to unassign me and reassign the issue to someone else if there’s interest. I’ll revisit this later and open a new PR if I’m able to fix it. Thanks for your support!

**Comment by MisRob** [comment]
Thanks for letting us know @iamshobhraj
